src/connectors/kessel/impl.js

```diff
@@ -0,0 +1,199 @@
+'use strict';
+
+const assert = require('assert');
+const Connector = require('../Connector');
+const log = require('../../util/log');
+
+const metrics = require('../metrics');
+
+// Import new Kessel SDK with ClientBuilder
+let ClientBuilder, fetchOIDCDiscovery, OAuth2ClientCredentials, OAuth2AuthRequest, fetchDefaultWorkspace, Allowed
+try {
+    // Import the new ClientBuilder from the updated SDK
+    const kesselSdk = require('@project-kessel/kessel-sdk/kessel/inventory/v1beta2');
+    ClientBuilder = kesselSdk.ClientBuilder;
+    
+    const allowedSdk = require('@project-kessel/kessel-sdk/kessel/inventory/v1beta2/allowed');
+    Allowed = allowedSdk.Allowed;
+
+    const oAuthSdk = require('@project-kessel/kessel-sdk/kessel/auth');
+    fetchOIDCDiscovery = oAuthSdk.fetchOIDCDiscovery;
+    OAuth2ClientCredentials = oAuthSdk.OAuth2ClientCredentials;
+    OAuth2AuthRequest = oAuthSdk.oauth2AuthRequest;
```

Comment:
Super-annoying style comment you can totally disregard...

This is 100% fine but can be written more succinctly using destructuring:

```
const {ClientBuilder} = require('@project-kessel/kessel-sdk/kessel/inventory/v1beta2');
const {Allowed} = require('@project-kessel/kessel-sdk/kessel/inventory/v1beta2/allowed');
const {fetchOIDCDiscovery, OAuth2ClientCredentials, OAuth2AuthRequest} = require('@project-kessel/kessel-sdk/kessel/auth');
const {fetchDefaultWorkspace} = require('@project-kessel/kessel-sdk/kessel/rbac/v2');
```